src/unity/python/turicreate/visualization/plot.py

```diff
@@ -7,22 +7,99 @@
 import json as _json
 
 class Plot(object):
+    """
+    An immutable, object representative of a visualization.
 
+    Notes
+    -----
+    - A plot object is returned through a proxy via the SFrame's or SArray's .plot() method
+
+    Examples
+    --------
+    Suppose 'plt' is an Plot Object
+
+    We can view it using:
+
+    >>> plt.show()
+
+    We can save it using:
+
+    >>> plt.save('vega_spec.json')
+
+    We can also save the vega representation of the plot without data:
+
+    >>> plt.save('vega_spec.json', False)
+
+    """
     def __init__(self, _proxy=None):
         if (_proxy):
             self.__proxy__ = _proxy
         else:
             self.__proxy__ = None
 
     def show(self):
-        import sys
-        import os
-        if sys.platform != 'darwin' and sys.platform != 'linux2':
-             raise NotImplementedError('Visualization is currently supported only on macOS and Linux.')
+        """
+        A method for displaying the Plot object
+
+        Notes
+        -----
+        - In a Jupyter Notebook the .show() method displays an inline plot
+        - In any other environment, the .show() method displays a native GUI
+
+        Examples
+        --------
+        Suppose 'plt' is an Plot Object
+
+        We can view it using:
 
-        self.__proxy__.get('call_function', {'__function_name__': 'show'})
+        >>> plt.show()
 
-    def save_vega(self, filepath, include_data=True):
+        """
+        display = False
+        try:
+            if(get_ipython().__class__.__name__ == "ZMQInteractiveShell"):
+                self._repr_javascript_()
+                display = True
+            else:
+                pass
+        except:
```

Comment:
It would be better to catch the specific exception that happens when we're not in a Jupyter Notebook (and allow other exceptions to get raised from here).